Lib/gftools/builder/__init__.py

```diff
@@ -0,0 +1,341 @@
+* ``buildWebfont``: Build WOFF2 fonts. Defaults to ``$buildStatic``.
+* ``outputDir``: Where to put the fonts. Defaults to ``../fonts/``
+* ``vfDir``: Where to put variable fonts. Defaults to ``$outputDir/variable``.
+* ``ttDir``: Where to put TrueType static fonts. Defaults to ``$outputDir/ttf``.
+* ``otDir``: Where to put CFF static fonts. Defaults to ``$outputDir/otf``.
+* ``woffDir``: Where to put WOFF2 static fonts. Defaults to ``$outputDir/webfonts``.
+* ``autohintTTF`: Whether or not to autohint TTF files. Defaults to ``true``.
+* ``axisOrder``: STAT table axis order. Defaults to fvar order.
+* ``familyName``: Family name for variable fonts. Defaults to family name of first source file.
+
+"""
+
+from fontTools.ttLib import TTFont
+from fontmake.font_project import FontProject
+from ufo2ft import CFFOptimization
+from gftools.fix import fix_font
+from gftools.stat import gen_stat_tables
+from fontTools.otlLib.builder import buildStatTable
+import statmake.classes
+import statmake.lib
+from babelfont import Babelfont
+import sys
+import os
+import shutil
+import glyphsLib
+import tempfile
+from ttfautohint.options import parse_args as ttfautohint_parse_args
+from ttfautohint import ttfautohint
+from fontTools.ttLib.woff2 import main as woff2_main
+import logging
+import yaml
+
+
+class GFBuilder:
+    def __init__(self, configfile=None, config=None):
+        if configfile:
+            self.config = yaml.load(open(configfile), Loader=yaml.SafeLoader)
+            if os.path.dirname(configfile):
+                os.chdir(os.path.dirname(configfile))
+        else:
+            self.config = config
+        self.masters = {}
+        self.logger = logging.getLogger("GFBuilder")
+        self.fill_config_defaults()
+
+    def build(self):
+        loglevel = getattr(logging, self.config["logLevel"].upper())
+        self.logger.setLevel(loglevel)
+        # Shut up irrelevant loggers
+        if loglevel != logging.DEBUG:
+            for irrelevant in [
+                "glyphsLib.classes",
+                "glyphsLib.builder.components",
+                "ufo2ft",
+                "ufo2ft.filters",
+                "ufo2ft.postProcessor",
+                "fontTools.varLib",
+                "cu2qu.ufo",
+                "glyphsLib.builder.builders.UFOBuilder",
+            ]:
+                logging.getLogger(irrelevant).setLevel(logging.ERROR)
+        if self.config["buildVariable"]:
+            self.build_variable()
+        if self.config["buildStatic"]:
+            self.build_static()
+
+    def load_masters(self):
+        if self.masters:
+            return
+        for src in self.config["sources"]:
+            if src.endswith("glyphs"):
+                gsfont = glyphsLib.GSFont(src)
+                self.masters[src] = []
+                for master in gsfont.masters:
+                    self.masters[src].append(Babelfont.open(src, master=master.name))
+            elif src.endswith("designspace"):
+                # XXX
```

Comment:
I take it this isn't implemented yet?

Reply:
If it doesn't work with designspaces yet, I don't mind if the first release is for .glyphs only. However, I'd like it to raise a `NotImplementedError` if a user supplies a designspace or we alter argv/argparse so it cannot accept a designspace.

Reply:
It should work with designspaces. This code is just auto-guessing the familyName if the user doesn't supply it. I'll make that clearer.